src/WebJobs.Extensions.Http/HttpTriggerAttributeBindingProvider.cs

```diff
@@ -213,12 +189,62 @@ internal static void ApplyBindingData(object target, IDictionary<string, object>
                 }
             }
 
+            /// <summary>
+            /// Gets the static strongly typed binding data contract
+            /// </summary>
+            internal Dictionary<string, Type> GetBindingDataContract(HttpTriggerAttribute attribute, ParameterInfo parameter)
+            {
+                // add contract members for POCO binding properties if binding to a POCO
+                var aggregateDataContract = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
+                if (_isUserTypeBinding && _bindingDataProvider?.Contract != null)
+                {
+                    aggregateDataContract.AddRange(_bindingDataProvider.Contract);
+                }
+
+                // add contract members for any route parameters
+                // query parameters 
+                if (!string.IsNullOrEmpty(attribute.Route))
+                {
+                    var routeParameters = _httpRouteFactory.GetRouteParameters(attribute.Route);
+                    var parameters = ((MethodInfo)parameter.Member).GetParameters().ToDictionary(p => p.Name, p => p.ParameterType, StringComparer.OrdinalIgnoreCase);
+                    foreach (string parameterName in routeParameters)
+                    {
+                        // don't override if the contract already includes a name
+                        if (!aggregateDataContract.ContainsKey(parameterName))
+                        {
+                            // if there is a method parameter mapped to this parameter
+                            // derive the Type from that
+                            Type type;
+                            if (!parameters.TryGetValue(parameterName, out type))
+                            {
+                                type = typeof(string);
+                            }
+                            aggregateDataContract[parameterName] = type;
+                        }
+                    }
+                }
+
+                // add headers collection to the contract
+                if (!aggregateDataContract.ContainsKey(HttpHeadersKey))
+                {
+                    aggregateDataContract.Add(HttpHeadersKey, typeof(IDictionary<string, string>));
+                }
+
+                // add query parameter collection to binding contract
+                if (!aggregateDataContract.ContainsKey(HttpQueryKey))
+                {
+                    aggregateDataContract.Add(HttpQueryKey, typeof(IDictionary<string, string>));
+                }
+
+                return aggregateDataContract;
+            }
+
             internal static async Task<IReadOnlyDictionary<string, object>> GetRequestBindingDataAsync(HttpRequestMessage request, Dictionary<string, Type> bindingDataContract = null)
```

Comment:
Our Node conversion logic does this same HttpRequest-->JObject. Can these paths share?